src/services/fileStorage/proxy-service.js

```diff
@@ -165,46 +170,52 @@ const fileStorageService = {
 			.then((allowedFiles) => {
 				const files = allowedFiles.filter(f => f);
 				if (!files.length) {
-					return new Forbidden();
+					return Promise.reject(new Forbidden());
 				}
 
 				return files;
 			})
 			.catch((e) => {
 				logger.error(e);
-				return Promise.reject(new Forbidden());
+				return Promise.reject(e);
 			});
 	},
 
 	/**
-     * @param params, contains storageContext and fileName in query
-     * @returns {Promise}
-     */
+	 * @param id, Object-ID of file to be removed (optional)
+	 * @param requestData, contains query parameters and userId/storageType set by middleware
+	 * @returns {Promise}
+	 */
 	remove(id, { query, payload }) {
 		const { userId, fileStorageType } = payload;
-		const { _id } = query;
+		const { _id = id } = query;
 		const fileInstance = FileModel.findOne({ _id });
 
-		return canDelete(userId, _id)
-			.then(() => fileInstance.exec())
+		return fileInstance.exec()
 			.then((file) => {
-				if (!file) return Promise.resolve({});
+				if (!file) {
+					return Promise.reject(new NotFound());
+				}
 
-				return createCorrectStrategy(fileStorageType).deleteFile(userId, file.storageFileName);
+				return Promise.all([
+					file,
+					canDelete(userId, _id).catch(() => Promise.reject(new Forbidden())),
+				]);
 			})
+			.then(([file]) => createCorrectStrategy(fileStorageType).deleteFile(userId, file.storageFileName))
 			.then(() => fileInstance.remove().exec())
 			.catch((e) => {
 				logger.error(e);
```

Comment:
however here we should have a more detailed error logging to remove lost data later